config.default.json

```diff
@@ -25,10 +25,30 @@
     },
     "apprisk": {
       "validations": [
+        {
+          "url": "https://$CHECKMARX/cxrestapi/auth/identity/connect/token",
+          "body": {
+            "BROKER_VAR_SUB": ["username", "password"],
+            "username": "${CHECKMARX_USERNAME}",
+            "password": "{CHECKMARX_PASSWORD}",
+            "grant_type": "password",
+            "scope": "sast_rest_api",
+            "client_id": "resource_owner_client",
+            "client_secret": "014DF517-39D1-4453-B7B3-9930C563627C"
+          },
+          "method": "post",
+          "headers": {
+            "Content-Type": "application/x-www-form-urlencoded"
+          }
+        }
       ],
       "default": {
       },
       "required": {
+        "CHECKMARX": "checkmarx.customer.com",
```

Comment:
@aarlaud actually i dont think we can required any field since the user could add any chosen integration from the list - should we add it to default?

Reply:
the field required here means that we won't care about the value (we might use it as placeholder/example), but will indicate to the broker that the connection details must have this field define.
It's not completely baked in yet, and based on your use case we're going to have to have a conditional logic to include things whenever applicable, ignore otherwise. We'll combine the 2 I think to achieve the desired behavior.